.github/workflows/publish.yml

```diff
@@ -0,0 +1,159 @@
+          token: ${{ secrets.ACTIONS_ACCESS_TOKEN  }}
+          ref: ${{ inputs.tag }}
+      - name: Install the latest version of rye
+        uses: eifinger/setup-rye@v2
+        with:
+          version: '0.29.0'
+          enable-cache: true
+      - name: Rye Setup
+        run: |
+           rye config --set-bool behavior.use-uv=true
+      - name: Python setup
+        run: rye pin ${{ matrix.python-version }}
+      - uses: actions/download-artifact@v4
+        with:
+          name: build
+          path: ./dist
+      - name: Prepare package and environment
+        run: |
+          rye sync
+          rye run toml unset --toml-path pyproject.toml tool.rye.workspace
+          rye sync -f --update-all
+      - name: Integration Testing
+        env:
+          PYTEST_XDIST_AUTO_NUM_WORKERS: 32
+          LABELBOX_TEST_API_KEY: ${{ secrets[matrix.prod-key] }}
+          DA_GCP_LABELBOX_API_KEY: ${{ secrets[matrix.da-test-key] }}
+          LABELBOX_TEST_ENVIRON: prod
+        run: |
+          rye add labelbox --path ./$(find ./dist/ -name *.tar.gz) --sync
+          cd libs/labelbox
+          rm pyproject.toml
+          rye run pytest tests/integration
+      - name: Data Testing
+        env:
+          PYTEST_XDIST_AUTO_NUM_WORKERS: 32
+          LABELBOX_TEST_API_KEY: ${{ secrets[matrix.prod-key] }}
+          DA_GCP_LABELBOX_API_KEY: ${{ secrets[matrix.da-test-key] }}
+          LABELBOX_TEST_ENVIRON: prod
+        run: |
+          rye add labelbox --path ./$(find ./dist/ -name *.tar.gz) --sync --features data
+          cd libs/labelbox
+          rye run pytest tests/data
+  pypi-publish:   
+    runs-on: ubuntu-latest
+    needs: ['test-build']
+    environment: 
+      name: publish
+      url: 'https://pypi.org/project/labelbox/'
+    permissions:
+      # IMPORTANT: this permission is mandatory for trusted publishing
+      id-token: write
+    steps:
+      - uses: actions/download-artifact@v4
+        with:
+          name: build
+      - name: Publish package distributions to PyPI
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          packages-dir: artifact/
+  container-publish:
+    runs-on: ubuntu-latest
+    needs: ['test-build'] 
+    env:
+      CONTAINER_IMAGE: "ghcr.io/${{ github.repository }}"
+    steps:
+      - name: downcase CONTAINER_IMAGE
+        run: |
+          echo "CONTAINER_IMAGE=${CONTAINER_IMAGE,,}" >> ${GITHUB_ENV}
+  
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+  
+      - name: Log in to the Container registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+  
+      - name: Build and push
+        uses: docker/build-push-action@v5
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          push: true
+  
+          platforms: |
+            linux/amd64
+            linux/arm64
```

Comment:
this is just out of curiosity: why are we not builidng for windows (not part of this pr)